core/src/main/java/com/linecorp/armeria/client/retry/RetryDecision.java

```diff
@@ -36,7 +39,11 @@ public static RetryDecision retry(Backoff backoff) {
         if (backoff == Backoff.ofDefault()) {
             return DEFAULT;
         }
-        return new RetryDecision(requireNonNull(backoff, "backoff"));
+        return new RetryDecision(requireNonNull(backoff, "backoff"), null);
+    }
+
+    static RetryDecision retry(Backoff backoff, RequestHeaders headers) {
```

Comment:
How about renaming it to `redirect`. `retry` sounds like doing the same thing.
```suggestion
    static RetryDecision redirect(Backoff backoff, RequestHeaders headers) {
```

Reply:
`retry` seems fine there isn't anything redirect-specific in this method